openmetadata-ui/src/main/resources/ui/src/pages/DataInsightPage/DataInsightPage.component.tsx

```diff
@@ -116,45 +134,91 @@ const DataInsightPage = () => {
   };
 
   const handleTeamChange = (teams: string[] = []) => {
+    setTeamOptions((prev) => ({
+      ...prev,
+      selectedOptions: teams,
+    }));
     setChartFilter((previous) => ({
       ...previous,
       team: teams.length ? teams.join(',') : undefined,
     }));
   };
 
   const handleTeamSearch = async (query: string) => {
-    if (fetchTeamSuggestions) {
+    if (fetchTeamSuggestions && !isEmpty(query)) {
       try {
-        const response = await fetchTeamSuggestions(query, 5);
-        setTeamOptions(getTeamFilter(response.values));
+        const response = await fetchTeamSuggestions(query, PAGE_SIZE);
+        setTeamOptions((prev) => ({
+          ...prev,
+
+          options: getTeamFilter(response.values),
+        }));
       } catch (_error) {
         // we will not show the toast error message for suggestion API
       }
+    } else {
+      setTeamOptions((prev) => ({
+        ...prev,
+        options: prev.defaultOptions,
+      }));
+    }
+  };
+
+  const handleTierSearch = async (query: string) => {
+    if (query) {
+      setTierOptions((prev) => ({
+        ...prev,
+        options: prev.options.filter((value) =>
+          value.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+        ),
+      }));
+    } else {
+      setTeamOptions((prev) => ({
+        ...prev,
+        options: defaultTierOptions,
```

Comment:
Why we are setting `defaultTierOptions` to TeamOptions? Is this intentional?

Reply:
Thanks for catching, it was not intentional, I have updated code